.buildkite/commands/run-ui-tests.sh

```diff
@@ -26,8 +26,19 @@ install_swiftpm_dependencies
 echo "--- :keyboard: Connecting Hardware Keyboard"
 defaults write com.apple.iphonesimulator ConnectHardwareKeyboard -bool true
 
-echo "--- 🧪 Testing"
+echo "--- 🚀 Booting Simulator"
 xcrun simctl list >> /dev/null
+xcrun simctl boot "$DEVICE"
+
+# Wait for the simulator to be fully booted
+echo "--- ⏳ Waiting for Simulator to be Ready"
+while ! xcrun simctl list | grep "$DEVICE" | grep "Booted"; do
+  echo "Waiting for $DEVICE to boot..."
+  sleep 2
+done
+echo "--- ✅ Simulator is Ready"
```

Comment:
Nitpick: The `--- ` syntax will generate an expandable logs section. That can be confusing when there is no content in it, like in the "✅ Simulator is Ready" case.

Suggestion 1: Remove the `---` and leave the `echo` in the group started by `--- ⏳ Waiting for...`

```suggestion
echo "⏳ Waiting for Simulator to be Ready"
while ! xcrun simctl list | grep "$DEVICE" | grep "Booted"; do
  echo "Waiting for $DEVICE to boot..."
  sleep 2
done
echo "✅ Simulator is Ready"
```

Suggestion 2: Replace `---` with `~~~` to retain the group, but de-emphasize it 

```suggestion
echo "~~~ ⏳ Waiting for Simulator to be Ready"
while ! xcrun simctl list | grep "$DEVICE" | grep "Booted"; do
  echo "Waiting for $DEVICE to boot..."
  sleep 2
done
echo "~~~ ✅ Simulator is Ready"
```
E.g. see the first lines in this screenshot are lighter than the last

<img width="351" alt="image" src="https://github.com/user-attachments/assets/89965690-dfc0-4363-b33c-eaf28809c642">

Reply:
Option 1 would be my favorite, because all the steps are related to booting the sim and therefore can be in the same group.

Reply:
thanks for the review and suggestions @mokagio! had a look and i prefer option 1 too, i'll commit that suggestion